src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1447,12 +1448,12 @@ private void CreateOperationMethods(OpenApiUrlTreeNode currentNode, NetHttpMetho
             {
                 (_, true) => [],
                 (null, _) => operation.Responses!
-                                .Where(static x => !errorStatusCodes.Contains(x.Key) && x.Value.Content is not null)
+                                .Where(static x => !errorStatusCodes.Contains(x.Key) && x.Key != DefaultResponseIndicator && x.Value.Content is not null)
```

Comment:
we need to check it's not equal to 4XX or 5XX either.
I'd:
1. create another hashet that contains default, 4XX and 5XX
2. use that for the validation
3. optional, abstract the lamba so we don't define it twice